### PR TITLE
Ability to kill process in the task manager

### PR DESCRIPTION
**Pull request by Supercip971**

### CODE REVIEW

File: applications/task-manager/main.cpp [hunk 1]
```diff
@@ -34,8 +36,13 @@ class TaskManagerWindow : public Window
         new Button(toolbar, BUTTON_FILLED, Icon::get("plus"), "New task");
 
         auto cancel_task_button = new Button(toolbar, BUTTON_TEXT, Icon::get("close"), "Cancel task");
-        cancel_task_button->on(Event::ACTION, [](auto) {
-            dialog_message(Icon::get("close"), "Cancel task", "Are you sure about that ?", DIALOG_BUTTON_YES | DIALOG_BUTTON_NO);
+        cancel_task_button->on(Event::ACTION, [&](auto) {
+            if(dialog_message(Icon::get("close"), "Cancel task", "Are you sure about that ?", DIALOG_BUTTON_YES | DIALOG_BUTTON_NO) == DIALOG_BUTTON_YES){
+                if(_last_selected_pid!=0){
+
```
Comment (Keyboard-Slayer) [inline applications/task-manager/main.cpp:42]: Can you remove that extra line ?

File: applications/task-manager/main.cpp [hunk 2]
```diff
@@ -49,7 +56,11 @@ class TaskManagerWindow : public Window
         });
 
         _table_timer->start();
-
+        _table->on(EventType::MOUSE_BUTTON_PRESS, [&](auto) {
+            if (_table->selected() >= 0)
+            {
+               _last_selected_pid = _table_model->data(_table->selected(), 0).as_int();
+        }});
```
Comment (Keyboard-Slayer) [inline applications/task-manager/main.cpp:63]: You should  separate those closing braces

**Review by Keyboard-Slayer — CHANGES_REQUESTED** [review]

**Comment by sleepy-monax** [comment]
The logic for killing processes should be implemented in the TaskModel

**Comment by Supercip971** [comment]
updated !

File: applications/task-manager/main.cpp [hunk 3]
```diff
@@ -16,15 +15,15 @@ class TaskManagerWindow : public Window
...
+    uint32_t _last_selected_row;
     Table *_table;
     RefPtr<TaskModel> _table_model;
     OwnPtr<Timer> _table_timer;
 
 public:
     TaskManagerWindow() : Window(WINDOW_RESIZABLE)
     {
-        _last_selected_pid = 0;
+        _last_selected_row = 0;
```
Comment (sleepy-monax) [inline applications/task-manager/main.cpp:26]: this variable is never used

**Review by sleepy-monax — CHANGES_REQUESTED** [review]

File: applications/task-manager/TaskModel.cpp [hunk 4]
```diff
@@ -137,3 +138,8 @@ String TaskModel::cpu_greedy()
 {
     return greedy(_data, "cpu");
 }
+
+void TaskModel::kill_task(int row)
+{
+   process_cancel( data(row, 0).as_int());
```
Comment (sleepy-monax) [inline applications/task-manager/TaskModel.cpp:144]: why not get the pid directly from _data? also you should do data(row, COLUMN_ID);

**Review by sleepy-monax — CHANGES_REQUESTED** [review]

**Comment by Supercip971** [comment]
re-fixed

**Comment by sleepy-monax** [comment]
Nice work 👍

**Pull request merged**